source/tests/common/dpmodel/test_dpa4_so3_gridnet.py

```
@@ -0,0 +1,473 @@
+
+def test_so3_deserialize_rejects_bad_projector() -> None:
+    """SO3GridNet.deserialize validates the nested projector @class/@version."""
+    _pt, dp_net = _build_so3_nets(mode="self", op_type="glu", layout="ndfc")
+    data = dp_net.serialize()
+    bad_class = copy.deepcopy(data)
+    bad_class["config"]["projector"]["@class"] = "S2GridProjector"
+    with pytest.raises(ValueError, match="projector"):
+        DPSO3GridNet.deserialize(bad_class)
+    bad_ver = copy.deepcopy(data)
+    bad_ver["config"]["projector"]["@version"] = 99
+    with pytest.raises(Exception):
```

Ruff (0.15.17)

[warning] Do not assert blind exception: `Exception`

(B017)